commit f3200d00ea42e485772ff92d6d649aa8eeb640c0
Author: Gleb Natapov <gleb@redhat.com>
Date:   Mon Dec 10 14:05:55 2012 +0200

    KVM: inject ExtINT interrupt before APIC interrupts
    
    According to Intel SDM Volume 3 Section 10.8.1 "Interrupt Handling with
    the Pentium 4 and Intel Xeon Processors" and Section 10.8.2 "Interrupt
    Handling with the P6 Family and Pentium Processors" ExtINT interrupts are
    sent directly to the processor core for handling. Currently KVM checks
    APIC before it considers ExtINT interrupts for injection which is
    backwards from the spec. Make code behave according to the SDM.
    
    Signed-off-by: Gleb Natapov <gleb@redhat.com>
    Acked-by: "Zhang, Yang Z" <yang.z.zhang@intel.com>
    Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>

diff --git a/arch/x86/kvm/i8259.c b/arch/x86/kvm/i8259.c
index 848206df0967..cc31f7c06d3d 100644
--- a/arch/x86/kvm/i8259.c
+++ b/arch/x86/kvm/i8259.c
@@ -241,6 +241,8 @@ int kvm_pic_read_irq(struct kvm *kvm)
 	int irq, irq2, intno;
 	struct kvm_pic *s = pic_irqchip(kvm);
 
+	s->output = 0;
+
 	pic_lock(s);
 	irq = pic_get_irq(&s->pics[0]);
 	if (irq >= 0) {